### PR TITLE
feat: added notification alert for notify button

### PR DESCRIPTION
**Pull request by ilasw**
## Changes
- Added Notification confirmation alert;
- Refactored `EnableNotification` component;

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://enable-notifications-for-notify-button.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/notifications/EnableNotification.tsx [hunk 1]
```diff
@@ -57,6 +57,133 @@ const sourceToButtonText: Partial<Record<NotificationPromptSource, string>> = {
   [NotificationPromptSource.SourceSubscribe]: 'Enable',
 };
 
+interface EnableNotificationAlertCTA {
```
Comment (sshanzel) [inline packages/shared/src/components/notifications/EnableNotification.tsx:60]: Following the [discussion here](https://dailydotdev.slack.com/archives/C07ATUPBPJ9/p1721655993518569), I think we can revert the changes to this file if the only reason was to support the banner.

File: packages/shared/src/components/sources/SourceActions/SourceActionsWithNotifyExperiment.tsx [hunk 2]
```diff
@@ -44,9 +48,11 @@ export const SourceActionsWithNotifyExperiment = ({
     source,
   });
 
+  const [isNotifyOpen, toggleNotifyOpen] = useToggle(false);
```
Comment (sshanzel) [inline packages/shared/src/components/sources/SourceActions/SourceActionsWithNotifyExperiment.tsx:51]: We should not need the state anymore and just use the existing hook to enable PNs 🥳

**Review by sshanzel — CHANGES_REQUESTED** [review]
Looks great overall. Just need to make some adjustments based on the discussion in Slack.

**Comment by ilasw** [comment]
@sshanzel I'm closing this for now as it should not be merged.
